Sources/PackageCollections/Storage/FilePackageCollectionsSourcesStorage.swift

```diff
@@ -10,13 +10,28 @@
 //
 //===----------------------------------------------------------------------===//
 
+import _Concurrency
 import Basics
 import Dispatch
 import struct Foundation.Data
 import class Foundation.JSONDecoder
 import class Foundation.JSONEncoder
 import struct Foundation.URL
 
+extension DispatchQueue {
+    func awaitingAsync<T>(_ closure: @escaping () throws -> T) async throws -> T {
+        try await withCheckedThrowingContinuation { continuation in
+            self.async {
+                do {
+                    try continuation.resume(returning: closure())
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+}
+
```

Comment:
All uses of this function are on `DispatchQueue.sharedConcurrent`, which makes it equivalent to `safe_async`.

```suggestion
```